### PR TITLE
chore: enable gradual typescript migration

### PR DESCRIPTION
**Pull request by suevalov**
* adds Typescript configuration to allow gradual migration
* changes Eslint configuration to be able to parse Typescript
* converts 2 files to Typescript
* `typings` are built and are located in `dist` folder, but are not exposed yet via `package.types` field

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -9,12 +14,20 @@ export default [
...
         babelrc: false,
         presets: [
           ['@babel/env', {
             modules: false
-          }]
+          }],
+          '@babel/typescript'
+        ],
+        plugins: [
+          '@babel/proposal-object-rest-spread'
```
Comment (marcolink) [inline rollup.config.js:30]: Do we need this in combination with TS?

Comment (suevalov) [reply rollup.config.js:30]: ah, probably not anymore. I was looking at rather old example

File: rollup.config.js [hunk 2]
```diff
@@ -9,12 +14,17 @@ export default [
       { file: pkg.main, format: 'cjs' }
     ],
     plugins: [
+      // Allows node_modules resolution
+      resolve({ extensions }),
+      // Compile TypeScript/JavaScript files
       babel({
+        extensions,
         babelrc: false,
```
Comment (marcolink) [inline rollup.config.js:22]: Didn't change in this PR, just wonder why we don't use the babelrc file, and declare things again in the rollup.config.

**Review by marcolink — APPROVED** [review]

File: rollup.config.js (hunk 2 above)
Comment (suevalov) [reply rollup.config.js:22]: `.babelrc` is used by test script and it has `rewire` plugin in it, but ideally we should have only one

**Review by anho — APPROVED** [review]

**Pull request merged**